modules/phase_field/include/materials/GammaAnisoGauss.h

```diff
@@ -0,0 +1,155 @@
+// 5D Gaussian anisotropy material object
+// Material properties adding anisotropy to gamma;
+// L is isotropic and computed from a given mobility.
+
+#pragma once
+
+#include "ADMaterial.h"
+#include "MooseMesh.h"
+#include "MathUtils.h"
+#include "GrainTrackerInterface.h"
+#include <vector>
+
+class GammaAnisoGauss : public ADMaterial
+{
+public:
+  static InputParameters validParams();
+  GammaAnisoGauss(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties() override;
+
+private:
+  void readQuaternionFile();
+  void readLibraryFile();
+
+  // Grains orientation quaternions.
+  std::vector<double> _qwR, _qxR, _qyR, _qzR;
+  // Normalized_Axis_X  Normalized_Axis_Y Normalized_Axis_Z  Angle(Radian) GB_Normal_X  GB_Normal_Y
+  // GB_Normal_Z  Minimun_GB_energy(J/m^2).
+  std::vector<double> _vbalibx, _vbaliby, _vbalibz, _thetabalib, _miubalibx, _miubaliby, _miubalibz,
+      _MinimaEnergy;
```

Comment:
Again, looks familiar. And think about your comments here. They should meaningfully add to the understanding. 
`_MinimaEnergy // Minimun_GB_energy(J/m^2)` is a little weak. 
and why not name `_vbalibx` to `_normalized_axis` (and combine the x,y,z variables into a single `RealVectorValue` variable)

Reply:
See updated files